go/vt/vttablet/tabletserver/planbuilder/plan.go

```diff
@@ -197,9 +200,10 @@ type MessageRowValues struct {
 
 // Plan is built for selects and DMLs.
 type Plan struct {
-	PlanID PlanType
-	Reason ReasonType
-	Table  *schema.Table
+	PlanID  PlanType
+	Reason  ReasonType
+	Table   *schema.Table
+	NewName *sqlparser.TableIdent
```

Comment:
TableIdent should be used as non-pointer variable. See other use cases and NewTableIdent. Consequently, you won't have to check for nil for enforcement. Just IsEmpty should be sufficient.